Neos.Flow/Classes/Mvc/Routing/Dto/UriConstraints.php

```diff
@@ -252,13 +254,13 @@ public function applyTo(UriInterface $templateUri, bool $forceAbsoluteUri): UriI
             $originalHost = $host = !empty($uri->getHost()) ? $uri->getHost() : $templateUri->getHost();
             $suffix = $this->constraints[self::CONSTRAINT_HOST_SUFFIX]['suffix'];
             $replaceSuffixes = $this->constraints[self::CONSTRAINT_HOST_SUFFIX]['replaceSuffixes'];
-            foreach ($replaceSuffixes as $replaceSuffix) {
-                if ($this->stringEndsWith($host, $replaceSuffix)) {
-                    $host = substr($host, 0, -strlen($replaceSuffix));
-                    break;
-                }
+
+            if ($replaceSuffixes === []) {
+                $host .= $suffix;
+            } else {
+                $regex = '/(' . implode('|', array_map('preg_quote', $replaceSuffixes)) . ')$/';
```

Comment:
Golden question: as this is not covered by tests, but the same expectation as above should be in place here. Add it?